cellpack/autopack/Ingredient.py

```diff
@@ -3241,86 +3340,81 @@ def checkSphCollisions(
             delta *= delta
             distA = numpy.sqrt(delta.sum(1))
-            # loop over all the grid points in the cube
-            # NOTE: used to filter these points to an inscribed sphere, but is expensive and didn't reduce points to check
-            for pti in range(len(pointsInCube)):
-                pt = pointsInCube[
+
+            for pti in range(len(pointsToCheck)):
+                pt = pointsToCheck[
                     pti
                 ]  # index of master grid point that is inside the sphere
                 distance_to_packing_location = distA[
                     pti
                 ]  # is that point's distance from the center of the sphere (packing location)
-
                 # distance is an array of distance of closest contact to anything currently in the grid
-                if (
-                    distance_to_packing_location <= radius_of_ing_being_packed
-                    and distance[pt] < -0.0001
-                ):
-                    # if there is an object encroaching into this packing area, that is
-                    # closer to the packing location than the radius of the object
-                    # we are trying to pack, probably a collision
-                    if level < self.maxLevel and (level + 1) < len(self.positions):
-                        nxtLevelSpheres = self.positions[level + 1]
-                        nxtLevelRadii = self.radii[level + 1]
+                collision = (
+                    distance[pt] + distance_to_packing_location
+                    <= radius_of_ing_being_packed
+                )
+
+                if collision:
+                    # an object is too close to the sphere at this level
+                    if not at_max_level:
+                        # if we haven't made it all the way down the sphere tree,
+                        # check a level down
+                        new_level = level + 1
+                        nxtLevelSpheres = self.positions[new_level]
+                        nxtLevelRadii = self.radii[new_level]
+                        # NOTE: currently with sphere trees, no children seem present
                         # get sphere that are children of this one
-                        ccenters = []
-                        cradii = []
-                        for sphInd in self.children[level][sphNum]:
-                            ccenters.append(nxtLevelSpheres[sphInd])
-                            cradii.append(nxtLevelRadii[sphInd])
-                        collision = self.checkSphCollisions(
-                            ccenters,
-                            cradii,
+                        # ccenters = []
+                        # cradii = []
+                        # for sphInd in self.children[level][sphNum]:
+                        #     ccenters.append(nxtLevelSpheres[sphInd])
+                        #     cradii.append(nxtLevelRadii[sphInd])
+                        return self.checkSphCollisions(
```

Comment:
it's import to return this recursive call so we can keep walking down the tree, instead of just calling it once.